yast2_rdp.pm

```diff
@@ -0,0 +1,194 @@
+        script_run 'gsettings set org.gnome.desktop.session idle-delay 0';
+    } else {
+        script_run("xscreensaver-command -exit");
+        if (check_var("DESKTOP", "lxde")) {
+            # Disable xscreensaver autostart on LXDE (from system wide config and user config)
+            script_sudo('sed -i "s/\@xscreensaver -no-splash//" /etc/xdg/lxsession/LXDE/autostart');
+            script_run('sed -i "s/\@xscreensaver -no-splash//" ~/.config/lxsession/LXDE/autostart');
+        }
+    }
+    wait_screen_change { send_key "ctrl-d" };
+    wait_still_screen;
+}
+
+sub check_service {
+    my ($type, $cmd) = @_;
+
+    x11_start_program('xterm', valid => 0);
+    becomeroot;
+    my $result = script_output("$cmd", proceed_on_failure => 1);
+    type_string "exit\n";
+    wait_screen_change { send_key "ctrl-d" };
+    wait_still_screen 2;
+
+    if ($type eq 'General') {
+        if ($result eq "active" or $result eq "enabled") {
+            return 1;
+        } else {
+            return 0;
+        }
+    } elsif ($type eq 'Firewall') {
+        if ($result eq 'ms-wbtX') {
+            return 1;
+        } elsif ($result eq 'X') {
+            return 0;
+        } else {
+            die "firewall command return unknown value";
+        }
+    } else {
+        die "The type of the command may be typo";
+    }
+}
+
+sub check_box {
+    my ($label, $tag) = @_;
+
+    if ($label eq 'RDP') {
+        assert_screen("$tag");
+        wait_screen_change { send_key "alt-n" };
+        wait_screen_change { send_key "alt-f" };
+        wait_still_screen;
+    } elsif ($label eq 'Firewall') {
+        assert_screen("$tag");
+        wait_screen_change { send_key "alt-o" };
+        wait_screen_change { send_key "alt-f" };
+        wait_still_screen;
+    } else {
+        die "$label is a unknown option in check_box";
+    }
+}
+
+sub run {
+    my $self = shift;
+    select_console("x11");
+
+    # work before test yast2-rdp
+    setup_system;
+
+    # yast2-rdp starts xrdp service. At this time, xrdp is NOT installed probably
+    $self->launch_yast2_module_x11("rdp", match_timeout => 120);
+    if (match_has_tag('yast2-rdp-ui-install-xrdp')) {
+        assert_and_click('yast2-rdp-ui-install-xrdp');
+        wait_screen_change { send_key "alt-i" };
+        wait_still_screen;
+    }
+    check_box('RDP', 'yast2-rdp-initial-after-installing-xrdp');
+
+    # check if xrdp service starts.
+    my $start  = check_service('General', 'echo $(systemctl is-active xrdp.service)');
+    my $enable = check_service('General', 'echo $(systemctl is-enabled xrdp.service)');
+    if (!($start and $enable)) {
+        die "It failed that yast2-rdp starts xrdp service";
+    }
+
+    # yast2-rdp stops xrdp service.
+    $self->launch_yast2_module_x11("rdp", match_timeout => 120);
+    check_box('RDP', 'yast2-rdp-ui-enable-xrdp');
+
+    # check if xrdp service stops.
+    $start  = check_service('General', 'echo $(systemctl is-active xrdp.service)');
```

Comment:
`systemctl` from utils.pm .....

Reply:
Thank you for your advice, I will have a try.